rust_snuba/rust_arroyo/src/processing/mod.rs

```diff
@@ -310,7 +312,7 @@ impl<TPayload: Clone + Send + Sync + 'static> StreamProcessor<TPayload> {
             }
 
             Err(StrategyError::Other(error)) => {
-                tracing::error!(error, "the thread errored");
+                return Err(RunError::Thread(error));
```

Comment:
I think this is a misnomer, the StreamProcessor doesn't know whether there was a thread at all

Reply:
Then maybe "the strategy errored"?

Reply:
sgtm